cfn-guard/src/lib.rs

```diff
@@ -377,6 +377,136 @@ fn apply_rule_operation(
+                        Some(match &rule.custom_msg {
+                            Some(c) => format!(
+                                "[{}] failed because [{}] is [{}] and {}",
+                                res_name,
+                                &rule.field,
+                                util::format_value(&val),
+                                c),
+                            None => format!(
+                                "[{}] failed because [{}] is [{}] and the permitted value is [> {}]",
+                                res_name,
+                                &rule.field,
+                                util::format_value(&val),
+                                rule_val.to_string()
+                            )
+                        }
+                        )
+                    } else {
+                        info!("Result: PASS");
+                        None
+                    }
+                }
+                _ => {
+                    error!("REQUIRE rule type that doesn't match RValueType of Regex, Variable or Value");
+                    None
+                }
+            }
+        }
+        enums::OpCode::LessThanOrEqualTo => {
+            match rule.rule_vtype {
+                enums::RValueType::Value | enums::RValueType::Variable => {
+                    if util::format_value(&val).parse::<f32>().unwrap() > util::strip_ws_nl(rule_val.to_string()).parse::<f32>().unwrap() {
+                        info!("Result: FAIL");
+                        Some(match &rule.custom_msg {
+                            Some(c) => format!(
+                                "[{}] failed because [{}] is [{}] and {}",
+                                res_name,
+                                &rule.field,
+                                util::format_value(&val),
+                                c),
+                            None => format!(
+                                "[{}] failed because [{}] is [{}] and the permitted value is [<= {}]",
+                                res_name,
+                                &rule.field,
+                                util::format_value(&val),
+                                rule_val.to_string()
+                            )
+                        }
+                        )
+                    } else {
+                        info!("Result: PASS");
+                        None
+                    }
+                }
+                _ => {
+                    error!("REQUIRE rule type that doesn't match RValueType of Regex, Variable or Value");
+                    None
+                }
+            }
+        }
+        enums::OpCode::GreaterThanOrEqualTo => {
+            match rule.rule_vtype {
+                enums::RValueType::Value | enums::RValueType::Variable => {
+                    if util::format_value(&val).parse::<f32>().unwrap() < util::strip_ws_nl(rule_val.to_string()).parse::<f32>().unwrap() {
```

Comment:
I think the operators for the LessThan/GreaterThan & LessThanOrEqualTo/GreaterThanOrEqualTo are reversed

Reply:
I read this in reverse.  I think all four of these would read more naturally if the operators were the same as the OpCode for the match arm.

For instance, `LessThanOrEqualTo` should use the `<=` rust numeric operator and have the first half of the `if` equal a pass.  In other words, fail it if it fails rust's numeric comparison. 

Ditto for all 4 cases.

Reply:
I was following the structure of the existing code. OpCode::Require is “!=“. I think, for consistency all comparisons should follow a similar structure.

Reply:
That's a fair point.  I think what we're seeing is that that initial decision to have the fail case be the first half of the `if-else` starts to get more and more confusing as we add more operators like these.

Let's do this:  You flip yours around here and I'll flip the other ones to make them consistent.